back/src/domains/marketing/adapters/establishmentMarketingGateway/BrevoEstablishmentMarketingGateway.ts

```diff
@@ -0,0 +1,358 @@
+        ENT_TYPE_PUBLIC_ACCUEILLIS: dto.isRegistered
+          ? searchableToTypePublic[dto.searchableBy]
+          : "",
+      },
+      updateEnabled: !!existingContact,
+    };
+
+    return this.#createContact(body).then((response) => {
+      if (existingContact ? 204 : 201) return;
+      throw new Error(
+        `Bad response with status '${
+          response.status
+        }' and body '${JSON.stringify(response.body)}'`,
+      );
+    });
+  }
+
+  async delete(contactEmail: Email): Promise<void> {
+    const existingContact = await this.getContactByEmail(contactEmail);
+
+    return !existingContact ||
+      (existingContact?.lists.length === 1 &&
+        existingContact?.lists.at(0) === this.#establishmentContactListId)
+      ? this.#deleteContactByEmail(contactEmail)
+      : this.#createContact({
+          attributes: {
+            EMAIL: contactEmail,
+            NOM: "",
+            PRENOM: "",
+            ENT_COMPTE_IC: "",
+            ENT_REFERENCE_SITE: "",
+            ENT_SIRET: "",
+            ENT_NB_CONVENTION: "",
+            ENT_EFFECTIF: "",
+            ENT_DATE_VALIDATION_DERNIERE_CONVENTION: "",
+            ENT_DATE_FIN_DERNIERE_CONVENTION: "",
+            ENT_DATE_PREM_CONVENTION: "",
+            ENT_CODE_DEPARTEMENT: "",
+            ENT_CODE_NAF: "",
+            ENT_DATE_DISPO: "",
+            ENT_LES_ENTREPRISES_SENGAGENT: "",
+            ENT_MAX_CONTACT_PER_WEEK: "",
+            ENT_NOMBRE_MER_RECUES: "",
+            ENT_NOMBRE_REPONSES_MER: "",
+            ENT_ROMES: "",
+            ENT_TYPE_PUBLIC_ACCUEILLIS: "",
+          },
+          email: contactEmail,
+          listIds: existingContact.lists.filter(
+            (listId) => listId !== this.#establishmentContactListId,
+          ),
+          updateEnabled: true,
+        }).then(() =>
+          this.#removeContactFromList({
+            email: contactEmail,
+            listId: this.#establishmentContactListId.toString(),
+          }),
+        );
+  }
+
+  #removeContactFromList({
+    email,
+    listId,
+  }: { email: Email; listId: string }): Promise<void> {
+    return this.#contactLimiter
+      .schedule(() =>
+        this.#httpClient.deleteContactFromList({
+          body: { emails: [email] },
+          headers: this.#brevoHeaders,
+          urlParams: { listId },
+        }),
+      )
+      .then((response) => {
+        if (response.status === 201) {
+          if (response.body.contacts.success.includes(email)) return;
+          if (response.body.contacts.failure.includes(email))
+            throw new Error(
+              `Remove contact '${email}' from list '${listId}' failed`,
+            );
+          throw new Error("Should not occurs.");
+        }
+        throw new Error(
+          `Bad response with status '${response.status}' and body '${response.body}'`,
+        );
+      });
+  }
+
+  async getContactByEmail(email: Email): Promise<BrevoContact> {
```

Comment:
```suggestion
  async getContactByEmail(email: Email): Promise<BrevoContact | undefined> {
```

L'intention est plus clair comme ça. (si ca vous va)